File: TreeHeight/TreeHeight/TreeHeight.cpp
#include<iostream>
using namespace std;
struct Node
{
	Node* left;
	Node* right;
	int data;
};
Node* newNode(int data) {
	Node* node = new Node();
	node->data = data;
	node->left = NULL;
	node->right = NULL;

	return node;
}
int maxHeight(Node* node) {
	if (node == NULL)	return 0;
	else {
		int leftHeight = maxHeight(node->left);
		int rightHeight = maxHeight(node->right);

		if (leftHeight>rightHeight) {
			return leftHeight + 1;
		}
		else{
			return rightHeight+1;
		}
	}
}
int main() {
	Node* node = newNode(1);

	node->left = newNode(2);
	node->right = newNode(3);
	node->left->left = newNode(4);
	node->left->right = newNode(5);

	cout << "max height of tree is " << maxHeight(node) << endl;

	system("pause");
	return 0;
}

File: py_TreeHeight/TreeHeight.py
class Node():
    def __init__(self,data):
        self.data=data;
        self.left=None
        self.right=None
def maxHeight(node):
    if node is None:
        return 0
    else:
        leftHeight=maxHeight(node.left)
        rightHeight=maxHeight(node.right)

        if (leftHeight>rightHeight):
            return leftHeight+1
        else:
            return rightHeight+1

root=Node(1)
root.left=Node(2)
root.right=Node(3)
root.left.left=Node(4)
root.left.right=Node(5)
print("max height of tree is %d"%(maxHeight(root)))